node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/InfrastructureProvisioner.java

```diff
@@ -1,113 +1,40 @@
 // Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 package com.yahoo.vespa.hosted.provision.maintenance;
 
-import com.yahoo.component.Version;
 import com.yahoo.config.provision.ApplicationId;
-import com.yahoo.config.provision.HostName;
-import com.yahoo.config.provision.HostSpec;
-import com.yahoo.config.provision.NodeType;
-import com.yahoo.config.provision.Provisioner;
+import com.yahoo.config.provision.InfraDeployer;
 import com.yahoo.log.LogLevel;
-import com.yahoo.transaction.Mutex;
-import com.yahoo.transaction.NestedTransaction;
-import com.yahoo.vespa.hosted.provision.Node;
 import com.yahoo.vespa.hosted.provision.NodeRepository;
-import com.yahoo.vespa.service.duper.DuperModel;
-import com.yahoo.vespa.service.monitor.DuperModelInfraApi;
-import com.yahoo.vespa.service.monitor.InfraApplicationApi;
 
 import java.time.Duration;
-import java.util.List;
-import java.util.Optional;
 import java.util.logging.Logger;
-import java.util.stream.Collectors;
 
 /**
- * This maintainer makes sure that 1. infrastructure nodes are allocated with correct wanted
- * version and 2. keeping {@link DuperModel} up to date. Source for the wanted version comes
- * from the target version set using /nodes/v2/upgrade/ endpoint.
+ * Periodically deploys infrastructure applications.
+ * TODO: Merge this with {@link PeriodicApplicationMaintainer}
```

Comment:
Currently this requires too many small hacks:
* `PeriodicApplicationMaintainer` is not regularly run in `test` environment for some reason
* `PeriodicApplicationMaintainer` depends on `Deployer::lastDeployTime` which is not implement for infra applications
* `ApplicationMaintainer` requires that there is at least 1 active node in the application, this is not the case in bootstrap phase for infra applications

Reply:
> not regularly run in test environment for some reason

That sounds strange. Not aware of any explicit disabling of that in test.

Reply:
Added in https://github.com/vespa-engine/vespa/pull/8571 @hmusum ?

Reply:
Because of the frequent application deletions there, and the config super models getting out of synch when this collided with deletions.